asserts/header_checks.go

```diff
@@ -166,3 +166,38 @@ func checkDigest(headers map[string]interface{}, name string, h crypto.Hash) ([]
 
 	return b, nil
 }
+
+func checkShouldBeString(headers map[string]interface{}, name string) (string, error) {
```

Comment:
Can we call this `checkOptionalString` and move it to the top, after the other *String ones?